From 31ba163efd5831eb9cd337b0d6cb222b1cb19c2d Mon Sep 17 00:00:00 2001
From: Brad Hall <brad@nicira.com>
Date: Sun, 2 Oct 2011 07:13:53 -0700
Subject: [PATCH] Change floating-snat to float-snat

With iptables v1.4.10 I'm running into an issue with the chain length of the
"nova-dhcpbridge-floating-snat" table (iptables it complaining that the name
is too long; 28 chars max).  This shortens the name of that table.

(cherry picked from commit 8700a97b701136e13c7ef959f13c89a3b7462fdb)

Change-Id: I00b1003aaa0b3771c29d6459d07023a8caa47706
---
 nova/network/linux_net.py |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/nova/network/linux_net.py b/nova/network/linux_net.py
index c0be962..a55ba2c 100755
--- a/nova/network/linux_net.py
+++ b/nova/network/linux_net.py
@@ -283,10 +283,10 @@ class IptablesManager(object):
         self.ipv4['nat'].add_rule('nova-postrouting-bottom', '-j $snat',
                                   wrap=False)
 
-        # And then we add a floating-snat chain and jump to first thing in
+        # And then we add a float-snat chain and jump to first thing in
         # the snat chain.
-        self.ipv4['nat'].add_chain('floating-snat')
-        self.ipv4['nat'].add_rule('snat', '-j $floating-snat')
+        self.ipv4['nat'].add_chain('float-snat')
+        self.ipv4['nat'].add_rule('snat', '-j $float-snat')
 
     @utils.synchronized('iptables', external=True)
     def apply(self):
@@ -459,7 +459,7 @@ def remove_floating_forward(floating_ip, fixed_ip):
 def floating_forward_rules(floating_ip, fixed_ip):
     return [('PREROUTING', '-d %s -j DNAT --to %s' % (floating_ip, fixed_ip)),
             ('OUTPUT', '-d %s -j DNAT --to %s' % (floating_ip, fixed_ip)),
-            ('floating-snat',
+            ('float-snat',
              '-s %s -j SNAT --to %s' % (fixed_ip, floating_ip))]
 
 
-- 
1.7.6.5

